File: service/src/main/java/com/iberianmotorsports/service/model/Race.java
package com.iberianmotorsports.service.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import org.hibernate.validator.constraints.Range;

import java.util.List;

@Entity
@Data
@Table(name = "RACE")
public class Race {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne()
    @JoinColumn(name = "championship_id", nullable = false)
    private Championship championship;

    @NotEmpty
    @Column(name = "track")
    private String track;

    @NotNull
    @Range(min=30)
    @Column(name = "pre_race_waiting_time_seconds")
    private Integer preRaceWaitingTimeSeconds;

    @NotNull
    @Column(name = "session_over_time_seconds")
    private Integer sessionOverTimeSeconds;

    @NotNull
    @Column(name = "ambient_temp")
    private Integer ambientTemp;

    @NotNull
    @Range(min=0, max=1)
    @Column(name = "cloud_level")
    private Float cloudLevel;

    @NotNull
    @Range(min=0, max=1)
    @Column(name = "rain")
    private Float rain;

    @NotNull
    @Range(min=0, max=7)
    @Column(name = "weather_randomness")
    private Integer weatherRandomness;

    @NotNull
    @Range(min=1)
    @Column(name = "post_qualy_seconds")
    private Integer postQualySeconds;

    @NotNull
    @Column(name = "post_race_seconds")
    private Integer postRaceSeconds;

    @NotEmpty
    @Column(name = "server_name")
    private String serverName;

    @OneToOne(mappedBy = "race", cascade = CascadeType.REMOVE, orphanRemoval = true)
    private RaceRules raceRules;

    @OneToMany(mappedBy = "race", cascade = CascadeType.REMOVE, orphanRemoval = true)
    private List<Session> sessionList;

    @Transient
    private Long championshipId;
}

File: frontend/src/app/model/Race.ts
import {RaceRules} from "./RaceRules";
import {Session} from "./Session";
import {Grid} from "./Grid";
import {Bop} from "./Bop";

export class Race {
    id: number;
    track: string;
    preRaceWaitingTimeSeconds: number;
    sessionOverTimeSeconds: number;
    ambientTemp: number;
    cloudLevel: number;
    rain: number;
    weatherRandomness: number;
    postQualySeconds: number;
    postRaceSeconds: number;
    serverName: string;
    sessionCount: number;
    startDate: string;
    raceRulesDTO: RaceRules;
    sessionDTOList: Session[];
    bopDTOList: Bop[];
    grid: Grid;
    championshipId: number;
    status: string;

    constructor() {
        this.id= -1;
        this.track= '';
        this.preRaceWaitingTimeSeconds= 150;
        this.sessionOverTimeSeconds= 150;
        this.ambientTemp= 19;
        this.cloudLevel= 0.35;
        this.rain=0.1;
        this.weatherRandomness= 2;
        this.postQualySeconds= 150;
        this.postRaceSeconds= 180;
        this.serverName= '';
        this.sessionCount= null;
        this.startDate = null;
        this.raceRulesDTO = null;
        this.sessionDTOList = [];
        this.bopDTOList = [];
        this.grid = null;
        this.championshipId = null;
        this.status = "PENDING";
    }
}